syntax = "proto3";
package video;

service ClientServerService {
  rpc Stream(stream ClientServerMessage) returns (stream ServerClientMessage);
}

message ClientServerMessage {
  oneof payload {
    ClientStartRequest client_start_request = 1;
    ClientStopRequest client_stop_request = 2;
    ClientPauseRequest client_pause_request = 3;
    ClientUnpauseRequest client_unpause_request = 4;
    ClientServerHeartbeat heartbeat = 5;
  }
}

message ServerClientMessage {
  oneof payload {
    ServerClientInfo info = 1;
    ServerClientChunk chunk = 2;
    ServerPauseRequest server_pause_request = 3;
    ServerUnpauseRequest server_unpause_request = 4;
    ServerClientHeartbeat heartbeat = 5;
    ServerNewUserJoinedRequest server_new_user_joined_request = 6;
  }
}

message ServerNewUserJoinedRequest {
  
}

message ClientServerHeartbeat {

}

message ServerClientHeartbeat {

}

message ServerPauseRequest {
  string timestamp = 1;
}

message ServerUnpauseRequest {
  string timestamp = 1;
}

message ClientStartRequest {
  string timestamp = 1;
}

message ClientStopRequest {
  string reason = 1;
}

message ClientPauseRequest {
  string timestamp = 1;
}

message ClientUnpauseRequest {
  string timestamp = 1;
}

message ServerClientChunk {
  bytes chunk = 1;
}

message ServerClientInfo {
  string info = 1;
}